r/configure

```diff
@@ -149,53 +150,42 @@ else
       if [ "$UNAME" = "Darwin" ] && [ "${OPENSSL_ROOT_DIR}" = "" ]; then
         brew --prefix openssl >/dev/null 2>&1
         if [ $? -eq 0 ]; then
-          OPENSSL_ROOT_DIR="`brew --prefix openssl`"; export OPENSSL_ROOT_DIR
+          export OPENSSL_ROOT_DIR="`brew --prefix openssl`"
+          export PKG_CONFIG_PATH="`brew --prefix openssl`/lib/pkgconfig${PKG_CONFIG_PATH:+:${PKG_CONFIG_PATH}}"
         fi
       fi
 
       if [ "${ARROW_DEPENDENCY_SOURCE}" = "" ]; then
-        ARROW_DEPENDENCY_SOURCE=AUTO; export ARROW_DEPENDENCY_SOURCE
+        export ARROW_DEPENDENCY_SOURCE=AUTO
       fi
       if [ "${ARROW_DEPENDENCY_SOURCE}" = "AUTO" ]; then
         pkg-config --version >/dev/null 2>&1
         if [ $? -ne 0 ]; then
-          export ARROW_DEPENDENCY_SOURCE=BUNDLED
-          echo "**** pkg-config not installed, setting ARROW_DEPENDENCY_SOURCE=BUNDLED"
+          echo "**** pkg-config is required to use automatically installed Apache Arrow C++"
```

Comment:
Why this change?

Reply:
But if dependency source is BUNDLED, we don't need pkg-config because all of the dependencies (aside from curl) are in `-larrow_bundled_dependences`, right?

My question isn't about whether we need pkg-config for source AUTO--that makes sense and was already the case--it is that you changed the default behavior to fail if pkg-config isn't found, rather than just using source BUNDLED.

Reply:
Ah, I understand. Sorry. I misunderstood the intent of this code.

I'll revert this change.